### PR TITLE
update modutils module

### PR DESCRIPTION
**Pull request by cgzones**
* deprecate old interfaces after insmod/depmod/updates-modules merge
* new handling of /run/tmpdiles.d
  use like /run with private types from 3rd party domains

### CODE REVIEW

**Comment by cgzones** [comment]
any comments on this one?

File: policy/modules/system/init.te [hunk 1]
```diff
@@ -289,7 +289,6 @@ ifdef(`init_systemd',`
 	udev_create_kobject_uevent_sockets(init_t)
 
 	optional_policy(`
-		systemd_relabelto_kmod_files(init_t)
```
Comment (pebenito) [inline policy/modules/system/init.te]: Why is this no longer needed?

File: policy/modules/system/modutils.if [hunk 2]
```diff
@@ -123,42 +123,83 @@ interface(`modutils_manage_module_config',`
...
 ## </summary>
 ## <param name="domain">
 ##	<summary>
 ##	Domain allowed to transition.
 ##	</summary>
 ## </param>
 #
-# cjp: this is added for pppd, due to nested
-# conditionals not working.
-interface(`modutils_domtrans_insmod_uncond',`
```
Comment (pebenito) [inline policy/modules/system/modutils.if]: This should be properly deprecated, not deleted.

File: policy/modules/system/modutils.if [hunk 3]
```diff
@@ -326,28 +339,6 @@ interface(`modutils_run_update_mods',`
...
-## <summary>
-##	Read kmod lib files.
-## </summary>
-## <param name="domain">
-##	<summary>
-##	Domain allowed access.
-##	</summary>
-## </param>
-#
-interface(`modutils_read_var_run_files',`
```
Comment (pebenito) [inline policy/modules/system/modutils.if]: This should be deprecated instead of deleted.

**Comment by cgzones** [comment]
updated with new `auth_use_pam_systemd` interface

**Comment by cgzones** [comment]
split up

**Comment by pebenito** [comment]
There are still questions in patch that need answers.

File: policy/modules/system/modutils.if (hunk 2 above)
Comment (cgzones) [inline policy/modules/system/modutils.if]: done

File: policy/modules/system/init.te (hunk 1 above)
Comment (cgzones) [inline policy/modules/system/init.te]: there is no type `systemd_kmod_conf_t` anymore, it's renamed to `kmod_tmpfiles_conf_t`, and i don't see why `init_t` should relabel files to it, cause of the new `systemd_tmpfiles_conf_filetrans` interface

**Comment by pebenito** [comment]
I tried pulling this and the relevant contrib change into my local checkout, and I still get deprecated interface usage.
```
policy/modules/contrib/dpkg.te:181: Warning: modutils_domtrans_insmod(dpkg_t) has been deprecated, please use modutils_domtrans() instead.
policy/modules/contrib/dpkg.te:316: Warning: modutils_domtrans_insmod(dpkg_script_t) has been deprecated, please use modutils_domtrans() instead.
policy/modules/system/init.te:1085: Warning: modutils_domtrans_insmod(initrc_t) has been deprecated, please use modutils_domtrans() instead.
policy/modules/system/iptables.te:123: Warning: modutils_domtrans_insmod(iptables_t) has been deprecated, please use modutils_domtrans() instead.
policy/modules/contrib/ncftool.te:78: Warning: modutils_domtrans_insmod(ncftool_t) has been deprecated, please use modutils_domtrans() instead.
policy/modules/contrib/portage.te:216: Warning: modutils_domtrans_insmod(portage_t) has been deprecated, please use modutils_domtrans() instead.
policy/modules/contrib/portage.te:216: Warning: modutils_run_update_mods(portage_t,portage_roles) has been deprecated, please use modutils_run() instead.
policy/modules/contrib/portage.te:216: Warning: modutils_domtrans_insmod(portage_t) has been deprecated, please use modutils_domtrans() instead.
policy/modules/contrib/rpm.te:353: Warning: modutils_domtrans_insmod(rpm_script_t) has been deprecated, please use modutils_domtrans() instead.
policy/modules/roles/sysadm.te:621: Warning: modutils_domtrans_insmod(sysadm_t) has been deprecated, please use modutils_domtrans() instead.
policy/modules/system/sysnetwork.te:206: Warning: modutils_domtrans_insmod(dhcpc_t) has been deprecated, please use modutils_domtrans() instead.
policy/modules/system/unconfined.te:114: Warning: modutils_domtrans_insmod(unconfined_t) has been deprecated, please use modutils_domtrans() instead.
policy/modules/contrib/usernetctl.te:72: Warning: modutils_domtrans_insmod(usernetctl_t) has been deprecated, please use modutils_domtrans() instead.
```

**Pull request merged**